<?xml version="1.0"?>
<launch>

  <arg name="map_file" default="$(maker_robot)/maps/layout.yaml"/>
  <arg name="model" default="$(maker_robot)/urdf/mobile_robot.urdf.xacro"/>
  
  <rosparam file="$(maker_robot)/config/controllers.yaml" command="load"/>
  <rosparam file="$(maker_robot)/config/joint_limits.yaml" command="load"/>
 
  <param name="robot_description" command="$(find xacro)/xacro.py $(arg model)" />
  
  <node name="robot_hardware_interface" pkg="mobile_robot_autonomous_navigation" type="mobile_robot_hardware_interface" output="screen">
    <remap from="/mobile_robot/mobile_base_controller/cmd_vel" to="/cmd_vel"/>
   </node>

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" >
  </node>
  
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(maker_robot)/config/nav_config.rviz"/>
  
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen"
        args="
            /mobile_robot/joints_update
            /mobile_robot/mobile_base_controller" >
  </node>



  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" >
  </node>
  
  <include file="$(find rplidar_ros)/launch/rplidar.launch" />
  <include file="$(maker_robot)/launch/amcl.launch" />
  <include file="$(maker_robot)/launch/move_base.launch" /> 
  
</launch>
